extensions/cornerstone/src/utils/stackSync/toggleStackImageSync.ts

```diff
@@ -1,90 +1,81 @@
-import calculateViewportRegistrations from './calculateViewportRegistrations';
-
-// [ {
-//   synchronizerId: string,
-//   viewports: [ { viewportId: string, renderingEngineId: string, index: number } , ...]
-// ]}
-let STACK_IMAGE_SYNC_GROUPS_INFO = [];
-
-export default function toggleStackImageSync({ toggledState, servicesManager, getEnabledElement }) {
+const disableSync = (syncName, servicesManager) => {
   const { syncGroupService, viewportGridService, displaySetService, cornerstoneViewportService } =
     servicesManager.services;
+  const viewports = getViewports(viewportGridService, displaySetService);
+  viewports.forEach(gridViewport => {
+    const { viewportId } = gridViewport.viewportOptions;
+    const viewport = cornerstoneViewportService.getCornerstoneViewport(viewportId);
+    if (!viewport) {
+      return;
+    }
+    syncGroupService.removeViewportFromSyncGroup(
+      viewport.id,
+      viewport.getRenderingEngine().id,
+      syncName
+    );
+  });
+};
 
-  if (!toggledState) {
-    STACK_IMAGE_SYNC_GROUPS_INFO.forEach(syncGroupInfo => {
-      const { viewports, synchronizerId } = syncGroupInfo;
-
-      viewports.forEach(({ viewportId, renderingEngineId }) => {
-        syncGroupService.removeViewportFromSyncGroup(viewportId, renderingEngineId, synchronizerId);
-      });
-    });
-
-    return;
-  }
-
-  STACK_IMAGE_SYNC_GROUPS_INFO = [];
-
-  // create synchronization groups and add viewports
-  const { viewports } = viewportGridService.getState();
+const getViewports = (viewportGridService, displaySetService) => {
```

Comment:
maybe rename to `_getReconstructableViewports` or something

Reply:
why are we even filetring the volume viewports? it is stackImageSync

Reply:
I'd rather not because eventually it should handle other types of viewports, once the CS3D side gets fixed to handle other types.